workspace/exercise.go

```diff
@@ -59,3 +64,53 @@ func (e Exercise) HasMetadata() (bool, error) {
 	}
 	return false, err
 }
+
+// HasLegacyMetadata checks for the presence of a legacy exercise metadata file.
+// If there is no such file, it could also be an unrelated directory.
+func (e Exercise) HasLegacyMetadata() (bool, error) {
+	_, err := os.Lstat(e.LegacyMetadataFilepath())
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// MigrationStatus represents the result of migrating a legacy metadata file.
+type MigrationStatus int
+
+// MigrationStatus
+const (
+	MigrationStatusErrorMkdir MigrationStatus = iota
+	MigrationStatusErrorRename
+	MigrationStatusErrorRemove
+	MigrationStatusNoop
+	MigrationStatusMigrated
+	MigrationStatusRemoved
+)
+
+// MigrateLegacyMetadataFile migrates a legacy metadata file to the modern location.
+// This is a noop if the metadata file isn't legacy.
+// If both legacy and modern metadata files exist, the legacy file will be deleted.
+func (e Exercise) MigrateLegacyMetadataFile() (MigrationStatus, error) {
+	if ok, _ := e.HasLegacyMetadata(); !ok {
+		return MigrationStatusNoop, nil
+	}
+	legacyMetadataFilepath := e.LegacyMetadataFilepath()
+	metadataFilepath := e.MetadataFilepath()
```

Comment:
Let's just call these inline. It's a couple of extra operations, potentially, but they're really cheap, and I'd rather not have to keep track of the local variables.